padertorch/contrib/examples/tasnet/tas_coders.py

```diff
@@ -127,3 +129,90 @@ def forward(self, w: torch.Tensor) -> torch.Tensor:
         """
         # Convolution from (B, N, T_enc) to (B, 1, T) and remove channel
         return self.decoder_1d(w)[:, 0, :]
+
+
+class StftEncoder(torch.nn.Module):
+    """
+    >>> mixture = torch.rand((2, 6, 203))
+    >>> stft_encoder = StftEncoder(N=258)
+    >>> encoded, num_frames = stft_encoder(mixture, [203, 150])
+    >>> encoded.shape
+    torch.Size([2, 6, 258, 20])
+    >>> num_frames
+    [20, 14]
+    >>> from paderbox.transform import stft
+    >>> import numpy as np
+    >>> stft_out = stft(\
+            mixture.numpy(), 256, 10, window_length=20, \
+            fading=None)
+    >>> stft_encoded = np.concatenate(\
+            [np.real(stft_out), np.imag(stft_out)], axis=-1\
+        ).transpose(0,1,3,2)
+    >>> np.testing.assert_allclose(stft_encoded, encoded, atol=1e-5)
+    """
+    def __init__(self, L: int = 20, N: int = 256, stride: int = None):
+        super().__init__()
+        self.L, self.N, self.stride = L, N, stride
+
+        if stride is None:
+            stride = L // 2
+        self.stft = STFT(size=N-2, shift=stride, window_length=L, fading=False,
+                         complex_representation='concat')
```

Comment:
Can you document, that you keep two uninformative values?
I think `N-2` is wrong. It may be something like: `(N - 1) // 2 * 2`

Reply:
N-2 is only wrong if we allow uneven stft sizes, which we do not support at the moment. But I can replace the size calculation and add an assert, that the feature_size is even.